Refactor StyleRuleCounterStyle for eliminating StyleProperties

vitorroriz opened the pull request:
#### b96c4a014f188f11c6fdfbb91937d42466fae41e
<pre>
Refactor StyleRuleCounterStyle for eliminating StyleProperties
<a href="https://bugs.webkit.org/show_bug.cgi?id=254524">https://bugs.webkit.org/show_bug.cgi?id=254524</a>
rdar://107267784

Reviewed by Antti Koivisto.

StyleRuleCounterStyle currently stores `StyleProperties` besides
`CSSCounterStyleDescriptors`.
This is used for facilitating serialization and read/write for the
CSSOM api. We can get rid of the `StyleProperties` member and use only
`CSSCounterStyleDescriptors` after we write a serializer for each
descriptor and adapt the setter function for setting only the
`CSSCounterStyleDescriptors` without havint to set the property that
originated the descriptor.

* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/counter-style-at-rule/prefix-suffix-syntax-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/counter-style-at-rule/speak-as-syntax-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/counter-style-at-rule/symbols-syntax-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/counter-style-prefix-suffix-syntax-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/counter-style-speak-as-syntax-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/counter-style-symbols-syntax-expected.txt:
- We and other browsers don&apos;t support url() as symbol for the
counter-style at-rule, so these should be failing. They were suceeding,
because we parsed it correctly, but since we don&apos;t support it, we
shouldn&apos;t be able to use this value for setting the descriptor.
- We don&apos;t support the speak-as descriptor, it should be failing.
- Note that these tests are duplicated. We should re-import WPT
and get rid of the dupes (rdar://107410996)

* Source/WebCore/css/CSSCounterStyle.cpp:
(WebCore::CSSCounterStyle::counterForSystemCyclic const):
(WebCore::CSSCounterStyle::counterForSystemFixed const):
(WebCore::CSSCounterStyle::counterForSystemSymbolic const):
(WebCore::CSSCounterStyle::counterForSystemAlphabetic const):
(WebCore::CSSCounterStyle::counterForSystemNumeric const):
(WebCore::CSSCounterStyle::counterForSystemAdditive const):
(WebCore::CSSCounterStyle::applyNegativeSymbols const):
(WebCore::CSSCounterStyle::applyPadSymbols const):
(WebCore::CSSCounterStyle::extendAndResolve):
- Since now Symbols is a struct, to support correct serialization of
custom-ident and string values, we should access the text value
with the .text member.

* Source/WebCore/css/CSSCounterStyle.h:
(WebCore::CSSCounterStyle::isExtendsUnresolved):
- Moving this flag to the descriptor so we can use it for serialization
of the system.

* Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
- We are renaming the translating functions, removing the &apos;translate&apos; prefix
to become in better compliance with WebKit style.
- The translation functions are now exposed via the header because when
a descriptor is set via CSSOM, we need to translate its CSSValue to the
descriptor value.
- Each descriptor has now a setter method, which will also take care of
its validation. Before, the validation would happen through setterInternal()
and newValueInvalidOrEqual() methods:
(WebCore::rangeFromStyleProperties):
(WebCore::rangeFromCSSValue):
(WebCore::symbolFromCSSValue):
(WebCore::nameFromCSSValue):
(WebCore::additiveSymbolsFromStyleProperties):
(WebCore::additiveSymbolsFromCSSValue):
(WebCore::padFromStyleProperties):
(WebCore::padFromCSSValue):
(WebCore::negativeSymbolsFromStyleProperties):
(WebCore::negativeSymbolsFromCSSValue):
(WebCore::symbolsFromStyleProperties):
(WebCore::symbolsFromCSSValue):
(WebCore::fallbackNameFromStyleProperties):
(WebCore::fallbackNameFromCSSValue):
(WebCore::prefixFromStyleProperties):
(WebCore::suffixFromStyleProperties):
(WebCore::extractSystemDataFromStyleProperties):
(WebCore::extractSystemDataFromCSSValue):
(WebCore::CSSCounterStyleDescriptors::setExplicitlySetDescriptors):
(WebCore::CSSCounterStyleDescriptors::create):
(WebCore::CSSCounterStyleDescriptors::areSymbolsValidForSystem):
(WebCore::CSSCounterStyleDescriptors::isValid const):
(WebCore::CSSCounterStyleDescriptors::setName):
(WebCore::CSSCounterStyleDescriptors::setSystemData):
(WebCore::CSSCounterStyleDescriptors::setNegative):
(WebCore::CSSCounterStyleDescriptors::setPrefix):
(WebCore::CSSCounterStyleDescriptors::setSuffix):
(WebCore::CSSCounterStyleDescriptors::setRanges):
(WebCore::CSSCounterStyleDescriptors::setPad):
(WebCore::CSSCounterStyleDescriptors::setFallbackName):
(WebCore::CSSCounterStyleDescriptors::setSymbols):
(WebCore::CSSCounterStyleDescriptors::setAdditiveSymbols):
(WebCore::CSSCounterStyleDescriptors::Symbol::cssText const):
(WebCore::CSSCounterStyleDescriptors::nameCSSText const):
(WebCore::CSSCounterStyleDescriptors::systemCSSText const):
(WebCore::CSSCounterStyleDescriptors::negativeCSSText const):
(WebCore::CSSCounterStyleDescriptors::prefixCSSText const):
(WebCore::CSSCounterStyleDescriptors::suffixCSSText const):
(WebCore::CSSCounterStyleDescriptors::rangesCSSText const):
(WebCore::CSSCounterStyleDescriptors::Pad::cssText const):
(WebCore::CSSCounterStyleDescriptors::padCSSText const):
(WebCore::CSSCounterStyleDescriptors::fallbackCSSText const):
(WebCore::CSSCounterStyleDescriptors::symbolsCSSText const):
(WebCore::CSSCounterStyleDescriptors::additiveSymbolsCSSText const):
(WebCore::translateRangeFromStyleProperties): Deleted.
(WebCore::symbolToString): Deleted.
(WebCore::translateAdditiveSymbolsFromStyleProperties): Deleted.
(WebCore::translatePadFromStyleProperties): Deleted.
(WebCore::translateNegativeSymbolsFromStyleProperties): Deleted.
(WebCore::translateSymbolsFromStyleProperties): Deleted.
(WebCore::translateFallbackNameFromStyleProperties): Deleted.
(WebCore::translatePrefixFromStyleProperties): Deleted.
(WebCore::translateSuffixFromStyleProperties): Deleted.
(WebCore::extractDataFromSystemDescriptor): Deleted.
(WebCore::CSSCounterStyleDescriptors::areSymbolsValidForSystem const): Deleted.
* Source/WebCore/css/CSSCounterStyleDescriptors.h:
(WebCore::CSSCounterStyleDescriptors::Symbol::operator== const):
(WebCore::CSSCounterStyleDescriptors::Symbol::operator!= const):
(WebCore::CSSCounterStyleDescriptors::setName): Deleted.
* Source/WebCore/css/CSSCounterStyleRule.cpp:

(WebCore::StyleRuleCounterStyle::StyleRuleCounterStyle):
(WebCore::StyleRuleCounterStyle::create):
- We no longer need to store the StyleProperties used for creating
the CSSCounterStyleDescriptors inside StyleRuleCounterStyle. Now we
store only CSSCounterStyleDescriptors.
(WebCore::CSSCounterStyleRule::CSSValueFromText):
(WebCore::CSSCounterStyleRule::setName):
(WebCore::CSSCounterStyleRule::setSystem):
(WebCore::CSSCounterStyleRule::setNegative):
(WebCore::CSSCounterStyleRule::setPrefix):
(WebCore::CSSCounterStyleRule::setSuffix):
(WebCore::CSSCounterStyleRule::setRange):
(WebCore::CSSCounterStyleRule::setPad):
(WebCore::CSSCounterStyleRule::setFallback):
(WebCore::CSSCounterStyleRule::setSymbols):
(WebCore::CSSCounterStyleRule::setAdditiveSymbols):
(WebCore::CSSCounterStyleRule::setSpeakAs):
(WebCore::symbolsValidForSystem): Deleted.
(WebCore::StyleRuleCounterStyle::newValueInvalidOrEqual const): Deleted.
(WebCore::StyleRuleCounterStyle::mutableProperties): Deleted.
(WebCore::CSSCounterStyleRule::setterInternal): Deleted.
- The CSSCounterStyleRule setters now use the setters of
CSSCounterStyleDescriptors to set each descriptor directly, without
having validating it with setterInternal() and newValueInvalidOrEqual(),
since each setter validates itself.
* Source/WebCore/css/CSSCounterStyleRule.h:
* Source/WebCore/css/parser/CSSParserImpl.cpp:
(WebCore::CSSParserImpl::consumeCounterStyleRule):

* Source/WebCore/rendering/RenderListMarker.cpp:
(WebCore::RenderListMarker::updateContent):
- Updating the access of symbol&apos;s via .text member, since it is now
a struct for supporting serialization (see first comments).

Canonical link: <a href="https://commits.webkit.org/262396@main">https://commits.webkit.org/262396@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/563b33ccdd515499e1047c181375c43f1a6834d4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1442 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1472 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1522 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2364 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/1271 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/1425 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1533 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1532 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/1399 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1455 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/1323 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1309 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2205 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1333 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/1293 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/1252 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1314 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/1332 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2400 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1367 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/1219 "Build is in progress. Recent messages:") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1285 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1300 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/357 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1409 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

anttijk commented on Source/WebCore/css/CSSCounterStyleRule.cpp:
```diff
@@ -220,104 +160,125 @@ void CSSCounterStyleRule::reattach(StyleRuleBase& rule)
     m_counterStyleRule = downcast<StyleRuleCounterStyle>(rule);
 }
 
+RefPtr<CSSValue> CSSCounterStyleRule::CSSValueFromText(CSSPropertyID propertyID, const String& valueText)
```
In WebKit coding style function names start with lowercase letter: `CSSCounterStyleRule::cssValueFromText`

anttijk commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -60,109 +69,167 @@ CSSCounterStyleDescriptors::Ranges translateRangeFromStyleProperties(const Style
+CSSCounterStyleDescriptors::Name nameFromCSSValue(RefPtr<CSSValue> value)
+{
+    return nameFromCSSValue(value.get());
+}
```
We probably don't need two versions of this function, just the RefPtr one.

anttijk commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -60,109 +69,167 @@ CSSCounterStyleDescriptors::Ranges translateRangeFromStyleProperties(const Style
+CSSCounterStyleDescriptors::AdditiveSymbols additiveSymbolsFromCSSValue(RefPtr<CSSValue> value)
+{
+    if (!value)
+        return { };
```
It should be sufficient to null test in one place only. If the caller is responsible then the `additiveSymbolsFromCSSValue` could take `Ref<CSSValue>` 

This comment applies to other similar functions in this patch.

anttijk commented on Source/WebCore/css/CSSCounterStyleRule.cpp:
```diff
@@ -220,104 +160,125 @@ void CSSCounterStyleRule::reattach(StyleRuleBase& rule)
+    auto systemData = extractSystemDataFromCSSValue(WTFMove(systemValue), system);
+    if (!mutableDescriptors().setSystemData(WTFMove(systemData)))
+        return;
+    CSSStyleSheet::RuleMutationScope mutationScope(this);
```
RuleMutationScope should be scoped over the mutation. That is the constructor should run before the mutation and destructor after (here and elsewhere).

anttijk reviewed (changes requested):
The RuleMutationScope issue needs to be fixed.

anttijk commented on Source/WebCore/css/CSSCounterStyleRule.h:
```diff
@@ -34,38 +34,33 @@
+    explicit StyleRuleCounterStyle(const AtomString&, CSSCounterStyleDescriptors&&);
 
     AtomString m_name;
-    // FIXME: we can get rid of m_properties and use only m_descriptors for storing the descriptors data (rdar://107267784).
-    Ref<StyleProperties> m_properties;
```
Nice!

anttijk commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -251,54 +303,243 @@ bool CSSCounterStyleDescriptors::areSymbolsValidForSystem() const
+    m_extendsName = systemData.first;
+    m_fixedSystemFirstSymbolValue = systemData.second;
+    return true;
 }
```
I don't think the return bools in these functions are used. The code could be simplified so these are just simple setters.

vitorroriz committed:
Refactor StyleRuleCounterStyle for eliminating StyleProperties
https://bugs.webkit.org/show_bug.cgi?id=254524
rdar://107267784

Reviewed by Antti Koivisto.

StyleRuleCounterStyle currently stores `StyleProperties` besides
`CSSCounterStyleDescriptors`.
This is used for facilitating serialization and read/write for the
CSSOM api. We can get rid of the `StyleProperties` member and use only
`CSSCounterStyleDescriptors` after we write a serializer for each
descriptor and adapt the setter function for setting only the
`CSSCounterStyleDescriptors` without havint to set the property that
originated the descriptor.

* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/counter-style-at-rule/prefix-suffix-syntax-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/counter-style-at-rule/speak-as-syntax-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/counter-style-at-rule/symbols-syntax-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/counter-style-prefix-suffix-syntax-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/counter-style-speak-as-syntax-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/counter-style-symbols-syntax-expected.txt:
- We and other browsers don't support url() as symbol for the
counter-style at-rule, so these should be failing. They were suceeding,
because we parsed it correctly, but since we don't support it, we
shouldn't be able to use this value for setting the descriptor.
- We don't support the speak-as descriptor, it should be failing.
- Note that these tests are duplicated. We should re-import WPT
and get rid of the dupes (rdar://107410996)

* Source/WebCore/css/CSSCounterStyle.cpp:
(WebCore::CSSCounterStyle::counterForSystemCyclic const):
(WebCore::CSSCounterStyle::counterForSystemFixed const):
(WebCore::CSSCounterStyle::counterForSystemSymbolic const):
(WebCore::CSSCounterStyle::counterForSystemAlphabetic const):
(WebCore::CSSCounterStyle::counterForSystemNumeric const):
(WebCore::CSSCounterStyle::counterForSystemAdditive const):
(WebCore::CSSCounterStyle::applyNegativeSymbols const):
(WebCore::CSSCounterStyle::applyPadSymbols const):
(WebCore::CSSCounterStyle::extendAndResolve):
- Since now Symbols is a struct, to support correct serialization of
custom-ident and string values, we should access the text value
with the .text member.

* Source/WebCore/css/CSSCounterStyle.h:
(WebCore::CSSCounterStyle::isExtendsUnresolved):
- Moving this flag to the descriptor so we can use it for serialization
of the system.

* Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
- We are renaming the translating functions, removing the 'translate' prefix
to become in better compliance with WebKit style.
- The translation functions are now exposed via the header because when
a descriptor is set via CSSOM, we need to translate its CSSValue to the
descriptor value.
- Each descriptor has now a setter method, which will also take care of
its validation. Before, the validation would happen through setterInternal()
and newValueInvalidOrEqual() methods:
(WebCore::rangeFromStyleProperties):
(WebCore::rangeFromCSSValue):
(WebCore::symbolFromCSSValue):
(WebCore::nameFromCSSValue):
(WebCore::additiveSymbolsFromStyleProperties):
(WebCore::additiveSymbolsFromCSSValue):
(WebCore::padFromStyleProperties):
(WebCore::padFromCSSValue):
(WebCore::negativeSymbolsFromStyleProperties):
(WebCore::negativeSymbolsFromCSSValue):
(WebCore::symbolsFromStyleProperties):
(WebCore::symbolsFromCSSValue):
(WebCore::fallbackNameFromStyleProperties):
(WebCore::fallbackNameFromCSSValue):
(WebCore::prefixFromStyleProperties):
(WebCore::suffixFromStyleProperties):
(WebCore::extractSystemDataFromStyleProperties):
(WebCore::extractSystemDataFromCSSValue):
(WebCore::CSSCounterStyleDescriptors::setExplicitlySetDescriptors):
(WebCore::CSSCounterStyleDescriptors::create):
(WebCore::CSSCounterStyleDescriptors::areSymbolsValidForSystem):
(WebCore::CSSCounterStyleDescriptors::isValid const):
(WebCore::CSSCounterStyleDescriptors::setName):
(WebCore::CSSCounterStyleDescriptors::setSystemData):
(WebCore::CSSCounterStyleDescriptors::setNegative):
(WebCore::CSSCounterStyleDescriptors::setPrefix):
(WebCore::CSSCounterStyleDescriptors::setSuffix):
(WebCore::CSSCounterStyleDescriptors::setRanges):
(WebCore::CSSCounterStyleDescriptors::setPad):
(WebCore::CSSCounterStyleDescriptors::setFallbackName):
(WebCore::CSSCounterStyleDescriptors::setSymbols):
(WebCore::CSSCounterStyleDescriptors::setAdditiveSymbols):
(WebCore::CSSCounterStyleDescriptors::Symbol::cssText const):
(WebCore::CSSCounterStyleDescriptors::nameCSSText const):
(WebCore::CSSCounterStyleDescriptors::systemCSSText const):
(WebCore::CSSCounterStyleDescriptors::negativeCSSText const):
(WebCore::CSSCounterStyleDescriptors::prefixCSSText const):
(WebCore::CSSCounterStyleDescriptors::suffixCSSText const):
(WebCore::CSSCounterStyleDescriptors::rangesCSSText const):
(WebCore::CSSCounterStyleDescriptors::Pad::cssText const):
(WebCore::CSSCounterStyleDescriptors::padCSSText const):
(WebCore::CSSCounterStyleDescriptors::fallbackCSSText const):
(WebCore::CSSCounterStyleDescriptors::symbolsCSSText const):
(WebCore::CSSCounterStyleDescriptors::additiveSymbolsCSSText const):
(WebCore::translateRangeFromStyleProperties): Deleted.
(WebCore::symbolToString): Deleted.
(WebCore::translateAdditiveSymbolsFromStyleProperties): Deleted.
(WebCore::translatePadFromStyleProperties): Deleted.
(WebCore::translateNegativeSymbolsFromStyleProperties): Deleted.
(WebCore::translateSymbolsFromStyleProperties): Deleted.
(WebCore::translateFallbackNameFromStyleProperties): Deleted.
(WebCore::translatePrefixFromStyleProperties): Deleted.
(WebCore::translateSuffixFromStyleProperties): Deleted.
(WebCore::extractDataFromSystemDescriptor): Deleted.
(WebCore::CSSCounterStyleDescriptors::areSymbolsValidForSystem const): Deleted.
* Source/WebCore/css/CSSCounterStyleDescriptors.h:
(WebCore::CSSCounterStyleDescriptors::Symbol::operator== const):
(WebCore::CSSCounterStyleDescriptors::Symbol::operator!= const):
(WebCore::CSSCounterStyleDescriptors::setName): Deleted.
* Source/WebCore/css/CSSCounterStyleRule.cpp:

(WebCore::StyleRuleCounterStyle::StyleRuleCounterStyle):
(WebCore::StyleRuleCounterStyle::create):
- We no longer need to store the StyleProperties used for creating
the CSSCounterStyleDescriptors inside StyleRuleCounterStyle. Now we
store only CSSCounterStyleDescriptors.
(WebCore::CSSCounterStyleRule::CSSValueFromText):
(WebCore::CSSCounterStyleRule::setName):
(WebCore::CSSCounterStyleRule::setSystem):
(WebCore::CSSCounterStyleRule::setNegative):
(WebCore::CSSCounterStyleRule::setPrefix):
(WebCore::CSSCounterStyleRule::setSuffix):
(WebCore::CSSCounterStyleRule::setRange):
(WebCore::CSSCounterStyleRule::setPad):
(WebCore::CSSCounterStyleRule::setFallback):
(WebCore::CSSCounterStyleRule::setSymbols):
(WebCore::CSSCounterStyleRule::setAdditiveSymbols):
(WebCore::CSSCounterStyleRule::setSpeakAs):
(WebCore::symbolsValidForSystem): Deleted.
(WebCore::StyleRuleCounterStyle::newValueInvalidOrEqual const): Deleted.
(WebCore::StyleRuleCounterStyle::mutableProperties): Deleted.
(WebCore::CSSCounterStyleRule::setterInternal): Deleted.
- The CSSCounterStyleRule setters now use the setters of
CSSCounterStyleDescriptors to set each descriptor directly, without
having validating it with setterInternal() and newValueInvalidOrEqual(),
since each setter validates itself.
* Source/WebCore/css/CSSCounterStyleRule.h:
* Source/WebCore/css/parser/CSSParserImpl.cpp:
(WebCore::CSSParserImpl::consumeCounterStyleRule):

* Source/WebCore/rendering/RenderListMarker.cpp:
(WebCore::RenderListMarker::updateContent):
- Updating the access of symbol's via .text member, since it is now
a struct for supporting serialization (see first comments).

Canonical link: https://commits.webkit.org/262396@main
